docs/source/contributing/development.md

```diff
@@ -0,0 +1,450 @@
+main library class.
+
+Some good examples with slightly different approaches:
+- Composition (has-a): [RPA.DocumentAI](../../../packages/main/src/RPA/DocumentAI/DocumentAI.py)
+- Inheritance (is-a): [RPA.Desktop](../../../packages/main/src/RPA/Desktop/__init__.py)
+
+So based on your needs, feel free to choose the simplest and most Pythonic one.
+
+#### Separate package (if needed)
+
+Some libraries are too havy on dependencies and the number of keywords to be
+implemented directly into [main](../../../packages/main), therefore we create a new
+package in this "monolith" repo, and we place them inside by following the same
+structure. Afterwards this becomes self-contained and gets released as a separate
+package into PyPI.
+
+Based on the frequency of usage (and slightly approachable  dependency train), such
+package might get automatically included in the _main_ one  by pinning it into the
+[pyproject.toml](../../../packages/main/pyproject.toml) file. If not, then for sure
+this has to be pinned in the _meta_'s [pyproject.toml](../../../pyproject.toml) in
+order to get documentation built and test how all our packages will go altogether.
+
+### Required changes
+
+Is not easy to know by heart all the changes that need to be done when improving or
+adding a new library, so here's a list with all the areas of interest:
+- The library module/package itself discussed above.
+- Updates into _pyproject.toml_:
+  - Its newly required dependencies should be added with `poetry add <package>` command.
+    (use `-d` option if is a development one)
+    - The new packages will be visible in the project's _pyproject.toml_ file and a
+      command like `inv install.install-updates` will ensure a fresh _poetry.lock_ file
+      after which the current _.venv_ is updated to.
+    - Use `poetry update -vvv` if you want to see the output during the long runs that
+      _meta_ building has.
+  - Bump the `<major>.<minor>.<hotfix>` version number appropriately to reflect the
+    type of change you introduced. (make sure you follow
+    [semantic versioning](https://semver.org/))
+    - The `inv build.version` can help you here with the bumping and unique pre-release
+      version management. (to avoid colision during testing)
+  - Update the [meta](../../../pyproject.toml) package as well to get a fresh
+    _poetry.lock_ file if you're planning to release soon, right after the merge.
+- The changes you've done should be summarized in a human-friendly way in the
+  [release notes](../../../docs/source/releasenotes.rst), more about in the
+  [**Release**](#release) section.
+- Adding a new library (additional requirements):
+  - Ensure the new `RPA.<Library>` is now visible in the [README](../../../README.rst)
+    as well (under **Libraries**).
+  - There's a documentation entry under [libraries](../../../docs/source/libraries) as
+    well. (follow the existing pattern)
+  - Add exclusion rule in [invocations](../../../invocations/docs.py) `EXCLUDE` list,
```

Comment:
I've never updated the exclude list and don't know why we would, so this may need more explaing.